Fix generators issue

deivid-rodriguez opened the pull request:
#### :tophat: What? Why?

This PR fixes a generator issue reported on gitter where `bootsnap` wouldn't be correctly added to the end application's `Gemfile`, and thus the generated application will fail to boot.

Tests are failing, but they're not failing for me locally when using https://github.com/decidim/docker/pull/29 (that preinstalls decidim 0.12 in the base image). So I would like to merge that PR and use it here to see if the build is fixed.

I will forward-port to master later.

#### :pushpin: Related Issues

- Related to decidim/docker#29.

#### :clipboard: Subtasks
- [x] Add `CHANGELOG` entry

### :camera: Screenshots (optional)
_None_.

mrcasals commented:
Nice, thanks! I'll ping @josepjaume on that other PR, see if he can unblock and merge it 😄

deivid-rodriguez commented:
Thanks, @mrcasals! I already implemented his suggestion in that other PR, so I think it should be fine :)

deivid-rodriguez commented:
@mrcasals This is still WIP, the actual fix for the issue is not pushed, I was trying to get a failing test first. Hopefully once the docker image is updated this starts failing.

deivid-rodriguez commented:
Any news on unblocking that PR by the way?

josepjaume commented:
PR unblocked! Apologies for the delay!

mrcasals commented:
@deivid-rodriguez it's failing, but I don't know if it's failing as you expected to... 😅

deivid-rodriguez commented:
Definitely not, generator specs still pass! So weird because they fail for me locally...

deivid-rodriguez commented:
Eureka! Got my failure! CircleCI caching was getting in the middle.

deivid-rodriguez commented:
I changed the target branch to master, rebased it, and converted in a follow up PR to #3756. It does the following:

* Add a changelog entry for the missing bootsnap fix.
* Fix another generator's issue, where optional gems were not being included.
* Add regression tests.
* Other minor related improvements.
